radio_beam/beam.py

```diff
@@ -245,29 +267,12 @@ def deconvolve(self, other):
         limit = np.min(axes)
         limit = 0.1*limit*limit
         
-        # two cases...
-
-        # ... failure
         if (alpha < 0) or (beta < 0) or (s < t):
-
-            # Note failure as an empty beam
-            new_major = 0.0
-            new_minor = 0.0
-            new_pa = 0.0
-
-            # Record that things failed
-            failed = True
-            
-            # Check if it is close to a point source
-            if ((0.5*(s-t) < limit) and
-                (alpha > -1*limit) and
-                (beta > -1*limit)):
-                pointlike = True
+            if failure_returns_pointlike:
```

Comment:
Revisiting waaaay later - I think the conditions here, `alpha < 0, beta < 0, or s < t` are all in the pointlike mode, not a genuine failure mode.  We might need to add back in some real failure modes, but I think I have to merge this because of some upstream (spectral-cube) needs.